### ISSUE TITLE
Suggestion d'amélioration: Pouvoir pré-remplir les champs à l'aide d'une query string

### ISSUE THREAD

**Issue by OlivierGrenoble** [LAB-MI/attestation-deplacement-derogatoire-q4-2020#13]
Bonjour,
Serait il possible de modifier le code pour qu'on puisse pré-remplir les champs à l'aide d'une query string?

Comme ca on pourrait préparer une URL de ce genre:
https://media.interieur.gouv.fr/deplacement-covid-19?field-firstname=Toto&field-lastname=Titi
la mémoriser sur notre téléphone et pouvoir ainsi générer plus facilement nos attestations.

NB: Je n'ai mis que 2 champs dans cet exemple mais on pourrait tous les pré-remplir.

Merci

**Comment by pixeye33** [comment]
Bonjour,

Surtout PAS : une query string est envoyée au serveur, cela gâche donc tout le principe de l'application qui est de fonctionner 100% en local (mais ça j'ai aucun doute que les mainteneurs le savent).

Par contre, utiliser des "URI fragment" est bien plus envisageable (tout ce qui est après le # n'est pas envoyé, et est accessible en javascript à la manière de https://www.unkm.fr/)

**Comment by pierregoutagny** [comment]
Si l'objectif est de pouvoir mémoriser sur son téléphone, alors cette proposition rejoint #4, et je vous renvoie à la [réponse](https://github.com/LAB-MI/attestation-deplacement-derogatoire-q4-2020/issues/4#issuecomment-719041247) que j'y ai fait, en particulier concernant LAB-MI/attestation-couvre-feu-covid-19#4

**Comment by OlivierGrenoble** [comment]
@pierregoutagny  J'ai utilisé cette méthode lors du 1er confinement mais elle était moyennement satisfaisante (certains champs ne sont pas mémorisés).

@pixeye33  Dommage pour la solution avec une Query String qui me semblait élégante (je ne conteste pas ce que vous dites sur la requête serveur). Je ne vois pas comment faire la même chose avec des "URI fragment".

**Comment by a2br** [comment]
Je m'apprétais à dire exactement la même chose que @OlivierGrenoble 
Par rapport aux URI fragments, n'ayant jamais travaillé avec je suppose que ça se récupère sous forme de string, qui peut ensuite être nettoyée (ex: `#nom:Doe,prenom:John` => `{nom:"Doe", prenom:"John"}`) et utilisée

**Comment by pierregoutagny** [comment]
@OlivierGrenoble avec la PR (du repo sur l'attestation couvre-feu) que j'ai citée, tous les champs étaient retenus (allez voir la démo proposée par l'auteur de la PR).
Cependant, je suis d'accord que la possibilité de charger les informations via l'URL avec les URI fragments, voire le PDF précédent (cf #31), serait bien plus flexible et pratique.

**Comment by OlivierGrenoble** [comment]
@pierregoutagny  J'ai regardé ce qu'il y a sur le lien auquel tu faisais référence (https://github.com/LAB-MI/attestation-couvre-feu-covid-19/pull/4). Si ma compréhension est bonne, il faut ce changement pour que tous les champs soient mémorisés. Est ce d'ores et déjà utilisable? J'ai l'impression qu'il faudrait que ce changement soit mergé, ce qui n'est pas encore le cas.

**Comment by MacKentoch** [comment]
100% d'accord avec @pierregoutagny : inutile de surcharger / sur-complexifier.

Franchement écrire quelque champs dans un formulaire une fois par jour

**Comment by tar-gezed** [comment]
Je pensais faire une feature qui stock dans le localStorage les infos a sauvegarder et qui set les date et heure de sortie par défaut à l'heure actuelle, je sais pas si c'est pertinent ?

**Comment by viincenb** [comment]
> Je pensais faire une feature qui stock dans le localStorage les infos a sauvegarder et qui set les date et heure de sortie par défaut à l'heure actuelle, je sais pas si c'est pertinent ?

Je suis d'accord, et c'est déjà en cours: https://github.com/LAB-MI/attestation-deplacement-derogatoire-q4-2020/pull/43 et https://github.com/LAB-MI/attestation-deplacement-derogatoire-q4-2020/pull/44

**Comment by jaythegeek** [comment]
Je pense que `localStorage` c'est le meilleur option, comme ca le user peu effacer par des cookies le data qui est sur leur appareil. Le form avec tous les input pourrais etre mis la dedans. (desolé pour le francais :/)

**Comment by OlivierGrenoble** [comment]
Savez vous quand il y aura une solution fonctionnelle permettant de mémoriser les champs qu'on a rentré?

**Comment by sosnut** [comment]
C'est tout à votre honneur en tant que développeurs de chercher, et trouver, des moyens de simplifier la tâche de l'utilisateur.
Il faut cependant tenir compte du besoin initial : faire chuter les contacts avec les personnes extérieures, pour sauver des vies.

C'est un avis personnel, mais je pense que ce besoin incite à laisser un peu de friction sous la forme de caractères à retaper, afin d'éviter de faciliter la tâche aux personnes qui génèrent une attestation numérique toutes les heures pour passer la journée dehors et voir tous leurs amis (pas une vue de l'esprit, malheureusement).

Une telle application a un impact en terme de vies humaines, je crois qu'une réflexion est nécessaire avant chaque mise à disposition de fonctionnalités.

**Comment by mistraloz** [comment]
@sosnut : trouver des moyens simple de compliquer la vie à tout le monde, ce sont les valeurs de la France. Soyons patriote et pourquoi pas dans ce cas ajouter un random pour que les champs se mélangent aléatoirement à chaque affichage ? Voir ajouter un jeu de piste avec les cases à cocher façon guichet numérique (allez cocher la case là-bas si quand vous avez coché la case ci-dessous elle était bleue, ou sinon indiquez le motif en toute lettre dans le courrier postal que nous vous adresserons généralement sous trois semaines). 

Tant pis si on doit verbaliser les personnes qui auront mal remplis le formulaire car son ergonomie était respectueuse de nos valeurs nationales.

**Comment by a2br** [comment]
Haha, je n'irais pas jusqu'à parler de valeurs nationales, mais plutôt de bon sens. Si quelque chose sert à X, il serait logique de rendre l'accomplissement de X meilleur, plus rapide, plus simple, etc... En tous cas, merci quand même d'avoir partagé votre avis, c'est toujours intéressant d'entendre des gens penser différement, ça ne me serait pas venu à l'esprit. Aussi, merci beaucoup @mistraloz, tu devrais faire une PR pour intégrer ces merveilleuses idées 😂

**Comment by pierregoutagny** [comment]
@sosnut quid des personnes en situation de handicap pour qui remplir un formulaire peut prendre encore plus de temps, mais qui ont quand même besoin de se rendre fréquemment chez un professionnel de santé, ou des gens qui s'occupent de leurs parents et qui pourraient avoir à sortir en urgence (pour motifs familiaux impérieux) ?
De toute façon, si ce générateur n'implémente pas d'autofill, des forks avec autofill seront (ou sont déjà) publiés. Je préfère que le générateur que les gens préfèrent utiliser soit celui-ci, plutôt que de savoir qu'il existe une version plus pratique et plus utilisée, mais pas nécessairement à jour ou fiable.
